pkg/scheduler/factory.go

```diff
@@ -228,33 +258,6 @@ func (c *Configurator) createFromConfig(policy schedulerapi.Policy) (*Scheduler,
 		}
 	}
 
-	var extenders []core.SchedulerExtender
```

Comment:
That's set in [`scheduler.go`](https://github.com/kubernetes/kubernetes/pull/88768/files#diff-0260dd7e154d411be493ada0753dc547R297), after the config has been decoded and before the call to `createFromConfig`. 

So in the event that [`options.Extenders`](https://github.com/kubernetes/kubernetes/pull/88768/files#diff-0260dd7e154d411be493ada0753dc547R268) is nil (ie, when using a policy config) this still sets the configurator field in the same area

Reply:
policy != componentConfig

Reply:
Oh I see. Then why did the test have to change? `New` should still be able to handle it.
Also, add the appropriate comments in `New`.

Reply:
> Then why did the integration test have to change?
it was failing before you pointed out that we weren't parsing the ignored resources correctly for policy config. Just reverted it and it passes now.

Reply:
Great :)

However, I think this line makes more sense in factory, so that policy is only processed in a single place. But this is personal taste. Other than that, let's just add a few clarification comments.